src/NServiceBus.Core/Pipeline/Incoming/InvokeHandlerTerminator.cs

```diff
@@ -27,7 +27,7 @@ await messageHandler
                     .ThrowIfNull()
                     .ConfigureAwait(false);
             }
-            catch (Exception e)
+            catch (Exception e) when (!(e is OperationCanceledException))
```

Comment:
Since this rethrows wouldn't it be ok to catch all exception since we're enriching the exception data and that seems to make sense for handlers throwing OCE as well?